pandas/_libs/src/ujson/python/objToJSON.c

```diff
@@ -1539,15 +1552,15 @@ void Object_beginTypeContext(JSOBJ _obj, JSONTypeContext *tc) {
         PyObject *is_null_obj = PyObject_CallMethod(obj,
```

Comment:
Can this not be replaced with the function you are introducing? Seems like it should work to keep logic consistent?